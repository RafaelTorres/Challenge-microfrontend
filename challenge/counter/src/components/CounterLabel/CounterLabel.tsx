import { FC } from 'react';

import styles from './styles.module.css';

export interface CounterLabelProps {
  count?: number;
}

const CounterLabel: FC<CounterLabelProps> = ({ count = 0 }) => <p className={styles['counter-label']}>{count}</p>;

export default CounterLabel;
