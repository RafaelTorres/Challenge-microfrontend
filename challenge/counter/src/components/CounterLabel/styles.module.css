.counter-label {
  border-radius: 50%;
  border: 1px solid var(--border-color, black);
  width: 7rem;
  height: 7rem;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 2rem;
}
