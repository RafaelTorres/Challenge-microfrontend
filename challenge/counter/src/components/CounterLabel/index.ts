import CounterLabel, { CounterLabelProps } from './CounterLabel';

export type { CounterLabelProps };

export default CounterLabel;
