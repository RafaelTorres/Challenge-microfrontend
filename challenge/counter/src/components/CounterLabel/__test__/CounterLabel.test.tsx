import { render, screen } from '@testing-library/react';
import CounterLabel, { CounterLabelProps } from '..';

const renderPage = (props?: Partial<CounterLabelProps>) => render(<CounterLabel {...props} />);

describe('Counter Label', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render the component', () => {
    const { container } = renderPage();

    const totalLabel = screen.queryByText(/0/);
    expect(totalLabel).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render the component with custom amount', () => {
    const { container } = renderPage({ count: 1 });

    const totalLabel = screen.queryByText(/1/);
    expect(totalLabel).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
