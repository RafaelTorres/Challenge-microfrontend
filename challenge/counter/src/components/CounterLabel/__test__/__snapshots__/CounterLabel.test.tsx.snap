// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Label should render the component 1`] = `
<div>
  <p
    class="counter-label"
  >
    0
  </p>
</div>
`;

exports[`Counter Label should render the component with custom amount 1`] = `
<div>
  <p
    class="counter-label"
  >
    1
  </p>
</div>
`;
